Add Czech (cs_CZ.UTF-8) translation

tkudlicka opened the pull request:
## Summary

This PR adds complete Czech language support for Ghostty based on #6840 by @calamitywoah, with improvements from community review feedback.

## Changes

- **Translation file**: Added `po/cs_CZ.UTF-8.po` with all 60 UI strings translated to Czech
- **Locale registration**: Added `cs_CZ.UTF-8` to `src/os/i18n_locales.zig`

## Translation Improvements

Based on feedback from #6840 reviewers (@ambrozf, @deepj, @Maneren, @pluiedev), the following improvements were applied:

1. **Terminal Title**: Uses "název" (title) instead of "nadpis" (heading) for better accuracy
2. **Split Directions**: Uses directional adverbs (nahoru/dolů/doleva/doprava) instead of locative case for more natural commands
3. **Natural Phrasing**: "Ponechte prázdné" instead of overly formal "Zanechte prázdné"
4. **Consistent Formality**: "Zapamatovat volbu" (formal infinitive) instead of informal "Zapamatuj si volbu"
5. **Proper Czech Terminology**: "ladicí verzi" instead of the anglicism "debugovací verzi"

## Credits

- Original translation: Luna (@calamitywoah)
- Review contributions: @ambrozf, @deepj, @Maneren, @pluiedev
- Additional improvements and locale registration: @tkudlicka

## References

Closes #6840

nenimitojedno committed:
Add Czech (cs_CZ.UTF-8) translation

This commit adds complete Czech language support for Ghostty, including:
- All 60 UI strings translated to Czech
- Proper Czech grammar and natural phrasing
- Locale registration in i18n_locales.zig

Based on PR #6840 by @calamitywoah with improvements from community feedback.

Translation improvements applied:
- Uses "název" (title) instead of "nadpis" (heading) for better accuracy
- Directional adverbs for split commands (nahoru/dolů/doleva/doprava)
- Natural phrasing: "Ponechte prázdné" instead of "Zanechte prázdné"
- Formal infinitive form: "Zapamatovat" instead of informal "Zapamatuj si"
- Proper Czech terminology: "ladicí" instead of anglicism "debugovací"

Credits:
- Original translation: Luna (@calamitywoah)
- Review contributions: @ambrozf, @deepj, @Maneren, @pluiedev
- Additional improvements: @tkudlicka

References #6840

00-kat commented on po/cs_CZ.UTF-8.po:
```diff
@@ -0,0 +1,320 @@
+"POT-Creation-Date: 2025-07-22 17:18+0000\n"
+"PO-Revision-Date: 2025-08-22 14:52+0300\n"
+"Last-Translator: reo101 <pavel.atanasov2001@gmail.com>\n"
+"Language-Team: Czech <@todo:add-email-here>\n"
```
This looks wrong, surely that isn't meant to be a todo.

00-kat reviewed (dismissed):
We hadn't yet documented this at the time #6840 was created, but please also [add your locale to the `CODEOWNERS` file](https://github.com/ghostty-org/ghostty/blob/38664afde0c17474c8f95cd38489d397a5930580/po/README_TRANSLATORS.md?plain=1#L150-L160)!

nenimitojedno committed:
Add Czech locale to CODEOWNERS

nenimitojedno committed:
cr fixes

tkudlicka commented on po/cs_CZ.UTF-8.po:
```diff
@@ -0,0 +1,320 @@
+"Report-Msgid-Bugs-To: m@mitchellh.com\n"
+"POT-Creation-Date: 2025-07-22 17:18+0000\n"
+"PO-Revision-Date: 2025-08-22 14:52+0300\n"
+"Last-Translator: reo101 <pavel.atanasov2001@gmail.com>\n"
```
Yeah, I copied it from Bulgarian but didn’t update it.

nenimitojedno committed:
Update Language-Team field to just 'Czech'

ambrozf commented on po/cs_CZ.UTF-8.po:
```diff
@@ -0,0 +1,320 @@
+"commands may be executed."
+msgstr ""
+"Vložení tohoto textu do terminálu může být nebezpečné, protože vypadá "
+"to, že mohou být spuštěny nějaké příkazy."
```
The structure of this sentence doesn't look correct. 

Possibly:
Vložení tohoto textu do terminálu může být nebezpečné, **protože se zdá**, že mohou být spuštěny nějaké příkazy.

nenimitojedno committed:
Fix sentence structure in unsafe paste warning

Use 'protože se zdá' instead of 'protože vypadá' for more natural Czech.

Suggested by @ambrozf

olevekavitocz commented on po/cs_CZ.UTF-8.po:
```diff
@@ -0,0 +1,320 @@
+
+#: src/apprt/gtk/ui/1.5/prompt-title-dialog.blp:5
+msgid "Change Terminal Title"
+msgstr "Změnit název terminálu"
```
neznám kontext -> nebude to spíš název okna terminálu?

olevekavitocz commented on po/cs_CZ.UTF-8.po:
```diff
@@ -0,0 +1,320 @@
+"and either reload your configuration or ignore these errors."
+msgstr ""
+"V konfiguraci byly nalezeny chyby. Prosím zkontrolujte chybový výpis níže "
+"a buď konfiguraci znovu načtěte, nebo chyby ignorujte."
```
".. a konfiguraci znovu načtěte, případně chyby ignorujte" ... buď je šílenost :-D

olevekavitocz commented on po/cs_CZ.UTF-8.po:
```diff
@@ -0,0 +1,320 @@
+#: src/apprt/gtk/ui/1.0/menu-surface-context_menu.blp:18
+#: src/apprt/gtk/ui/1.0/menu-window-titlebar_menu.blp:73
+msgid "Clear"
+msgstr "Vyčistit"
```
to je jako /clear ? maybe bych volil vymazat

olevekavitocz commented on po/cs_CZ.UTF-8.po:
```diff
@@ -0,0 +1,320 @@
+
+#: src/apprt/gtk/ui/1.0/menu-window-titlebar_menu.blp:85
+msgid "Command Palette"
+msgstr "Paleta příkazů"
```
maybe obráceně "Příkazová paleta" případně "Nabídka příkazů"

olevekavitocz commented on po/cs_CZ.UTF-8.po:
```diff
@@ -0,0 +1,320 @@
+#: src/apprt/gtk/ui/1.5/ccw-osc-52-read.blp:81
+#: src/apprt/gtk/ui/1.5/ccw-osc-52-write.blp:77
+msgid "Remember choice for this split"
+msgstr "Zapamatovat volbu pro toto rozdělení"
```
to je jenom pamatování že to je splitlé ? Pokud ano tak bych to nechal jen jako "Zapamatovat si rozdělení" nebo "Pamatovat ..."

olevekavitocz reviewed (commented):
jinak mě to přišlo ok :-) snad jsem pomohl

tkudlicka committed:
feat(po): improve Czech translations for clarity

Refine Czech translations in `cs_CZ.UTF-8.po` to enhance clarity and
consistency. Changes include adjustments to phrasing for better
readability and alignment with common usage.

- Updated phrasing for configuration reload instructions.
- Replaced "Vyčistit" with "Vymazat" for "Clear".
- Changed "Paleta příkazů" to "Příkazová paleta" for "Command Palette".
- Improved wording for terminal warning messages.

No functional changes, only translation improvements.
